ml_tools/trackdatabase.py

```diff
@@ -7,46 +7,39 @@
 Handles reading and writing tracks (or segments) to a large database.  Uses HDF5 as a backing store.
 
 """
-
+import h5py
 import os
 import logging
+import filelock
+import datetime
 from multiprocessing import Lock
-
-import h5py
-import tables  # required for blosc compression to work
 import numpy as np
 
-# default lock for safe database writes.
-#
-# note for multiprocessing this will need to be overwritten with a shared lock for each process.
-# which can be done via
-#
-# def init_workers(lock):
-#    trackdatabase.HDF5_LOCK = lock
-#
-# pool = multiprocessing.Pool(self.workers_threads, initializer=init_workers, initargs=(shared_lock,))
-
-HDF5_LOCK = Lock()
-
 
 class HDF5Manager:
     """ Class to handle locking of HDF5 files. """
 
+    LOCK_FILE = "/var/lock/classifier-hdf5.lock"
```

Comment:
A single lock file location could cause problems when the code is being run more than once a single machine. Could the lock file live next to the DB file?  Just append `.lock` to the DB filename.